tools/nni_cmd/nnictl_utils.py

```diff
@@ -373,6 +379,146 @@ def webui_url(args):
     nni_config = Config(get_config_filename(args))
     print_normal('{0} {1}'.format('Web UI url:', ' '.join(nni_config.get_config('webuiUrl'))))
 
+def local_clean(directory):
+    '''clean up local data'''
+    print_normal('cleaning up {0}'.format(directory))
+    try:
+        shutil.rmtree(directory)
+    except FileNotFoundError as err:
+        print_error('{0} does not exist!'.format(directory))
+    
+def remote_clean(nni_config):
+    '''clean up remote data'''
+    machine_list = nni_config.get_config('experimentConfig').get('machineList')
+    for machine in machine_list:
+        passwd = machine.get('passwd')
+        userName = machine.get('username')
+        host = machine.get('ip')
+        port = machine.get('port')
+        remote_dir = '/' + '/'.join(['tmp', 'nni', 'experiments', nni_config.get_config('experimentId')])
+        sftp = create_ssh_sftp_client(host, port, userName, passwd)
+        print_normal('cleaning up {0}'.format(host + ':' + str(port) + remote_dir))
+        remote_remove_directory(sftp, remote_dir)
+    
+def hdfs_clean(nni_config):
+    '''clean up hdfs data'''
+    host = nni_config.get_config('experimentConfig').get('paiConfig').get('host')	
+    user_name = nni_config.get_config('experimentConfig').get('paiConfig').get('userName')
+    hdfs_client = HdfsClient(hosts='{0}:80'.format(host), user_name=user_name, webhdfs_path='/webhdfs/api/v1', timeout=5)	
+    full_path = '/' + '/'.join([user_name, 'nni', 'experiments', nni_config.get_config('experimentId')])
+    print_normal('deleting {0} in hdfs'.format(full_path))
+    hdfs_client.delete(full_path, recursive=True)
+    output_dir = nni_config.get_config('experimentConfig').get('paiConfig').get('outputDir')
+    if output_dir:
+        pattern = re.compile('hdfs://(?P<host>([0-9]{1,3}.){3}[0-9]{1,3})(:[0-9]{2,5})?(?P<baseDir>/.*)?')
+        match_result = pattern.match(output_dir)
+        if match_result:
+            output_host = match_result.group('host')
+            output_directory = match_result.group('baseDir')
+            if output_host == host:
+                print_normal('deleting {0} in hdfs'.format(output_directory))
+                hdfs_client.delete(output_directory, recursive=True)
+
+def experiment_clean(args):
+    '''clean up the experiment data'''
+    nni_config = Config(get_config_filename(args))
+    while True:
+        print('INFO: clean up all data for experiment {0}'.format(args.id))
+        inputs = input('INFO: do you want to continue?[y/n]:')
+        if not inputs.lower() or inputs.lower() in ['n', 'no']:
+            print_normal('Exit!')
+            exit(0)
+        elif inputs.lower() not in ['y', 'n', 'yes', 'no']:
+            print_warning('please input Y or N!')
+        else:
+            break
+    #clean local data
+    home = str(Path.home())
+    local_dir = os.path.join(home, 'nni', 'experiments', str(args.id))
```

Comment:
you assume that the data is under ~/nni/experiments. I remembered that users can config the path where the data/log is stored.

Reply:
Fixed, use logDir first.